torch/nn/modules/transformer.py

```diff
@@ -321,25 +321,28 @@ def forward(self, src: Tensor, src_mask: Optional[Tensor] = None, src_key_paddin
         Shape:
             see the docs in Transformer class.
         """
+
+        # see Fig. 1 of https://arxiv.org/pdf/2002.04745v1.pdf
+
+        # self-attention block
+        def mha(x):
+            x = self.self_attn(x, x, x, attn_mask=src_mask, key_padding_mask=src_key_padding_mask)[0]
```

Comment:
nit: need_weights could be set to False if you're indexing at 0 for the return value

Reply:
Ah yeah nice catch!